tools/cli/src/commands/task_queue.rs

```diff
@@ -0,0 +1,84 @@
+use crate::{
+    args::{TargetEnvironment, TaskQueueArgs},
+    context::AppContext,
+};
+use anyhow::{Context, Result};
+use lavs_apis::id::TaskId;
+use lavs_task_queue::msg::{
+    ConfigResponse, CustomExecuteMsg, CustomQueryMsg, ExecuteMsg, QueryMsg, Requestor,
+};
+use layer_climb::{prelude::*, proto::abci::TxResponse};
+
+pub struct TaskQueue {
+    pub ctx: AppContext,
+    pub contract_addr: Address,
+}
+
+impl TaskQueue {
+    pub async fn new(ctx: AppContext, task_queue_args: &TaskQueueArgs) -> Result<Self> {
+        let addr_string = match task_queue_args.address.clone() {
+            Some(x) => x,
+            None => match ctx.args.target_env {
+                TargetEnvironment::Local => std::env::var("LOCAL_TASK_QUEUE_ADDRESS")
+                    .context("LOCAL_TASK_QUEUE_ADDRESS not found")?,
+                TargetEnvironment::Testnet => std::env::var("TEST_TASK_QUEUE_ADDRESS")
+                    .context("TEST_TASK_QUEUE_ADDRESS not found")?,
+            },
+        };
+
+        let contract_addr = ctx.chain_config.parse_address(&addr_string)?;
+
+        Ok(Self { ctx, contract_addr })
+    }
+
+    pub async fn add_task(
+        &self,
+        body: String,
+        description: String,
+        timeout: Option<u64>,
+    ) -> Result<(TaskId, TxResponse)> {
+        let payload = serde_json::from_str(&body).context("Failed to parse body into JSON")?;
+
+        let contract_config: ConfigResponse = self
+            .ctx
+            .get_client()
+            .await?
+            .querier
+            .contract_smart(
+                &self.contract_addr,
+                &QueryMsg::Custom(CustomQueryMsg::Config {}),
+            )
+            .await?;
+
+        let payment = match contract_config.requestor {
+            Requestor::OpenPayment(coin) => vec![new_coin(coin.amount, coin.denom)],
+            _ => Vec::new(),
+        };
+
+        let tx_resp = self
+            .ctx
+            .get_client()
+            .await?
+            .contract_execute(
+                &self.contract_addr,
+                &ExecuteMsg::Custom(CustomExecuteMsg::Create {
```

Comment:
These are serde flattened, so no need to wrap. `&CustomExecuteMsg::Create { ... }` will work the same.

I should rename that. (I know you made https://github.com/Lay3rLabs/avs-toolkit/issues/14 about that)

Reply:
hah, I didn't realize you could do that! I mean... I knew you could do it with the JSON representation, but it didn't click for me that you could do it from Rust too. TIL!

Reply:
done in https://github.com/Lay3rLabs/avs-toolkit/pull/29/commits/0680210ccbfa5e03e2163fd6c5e29f311ccc063f